### PR TITLE
De-flake fromPixels test failure

### LINKED ISSUE #6577: The myth of `HTMLVideoElement.readyState` and `HTMLVideoElement.requestVideoFrameCallback`
I'm seeing this error when running latest TFJS unit tests with the WebGPU backend:

```
Chrome 105.0.0.0 (Windows 10) webgpu backend webgpu webgpu {"WEBGPU_CPU_FORWARD":false} should behave well if WEBGPU_USER_IMPORT is true or false FAILED
 Error: The video element has not loaded data yet. Please wait for `loadeddata` event on the <video> element.
     at fromPixels_ (tfjs-core/src/ops/browser.ts:115:13 <- tfjs/tfjs-backend-webgpu/src/tfjs-backend-webgpu_test_bundle.js:5941:15)
     at Object.<anonymous> (tfjs-core/src/ops/browser.ts:279:10 <- tfjs/tfjs-backend-webgpu/src/tfjs-backend-webgpu_test_bundle.js:6021:14)
     at <Jasmine>
     at tfjs/tfjs-backend-webgpu/src/tfjs-backend-webgpu_test_bundle.js:52:64
     at <Jasmine>
     at __async (tfjs-backend-webgpu/src/tfjs-backend-webgpu_test_bundle.js:36:12)
     at Object.fromPixelsAsync (tfjs-backend-webgpu/src/tfjs-backend-webgpu_test_bundle.js:6004:12)
     at ../../tfjs-backend-webgpu/src/backend_webgpu_test.ts:248:36 <- tfjs/tfjs-backend-webgpu/src/tfjs-backend-webgpu_test_bundle.js:84990:47
     at <Jasmine>
     at fulfilled (tfjs-backend-webgpu/src/tfjs-backend-webgpu_test_bundle.js:39:27)
```

**OS:** Windows 10 (19044.1766)
**GPU:** Intel UHD 620 (driver version: 30.0.101.1994)
**Browser:** Chrome Canary (version: 105.0.5148.0)

What I don't understand is the readiness of the frame data has already been gated by `HTMLVideoElement.requestVideoFrameCallback`:

https://github.com/tensorflow/tfjs/blob/ca9799d16e23f7f31d4fdb15febb81d2eb4a2790/tfjs-backend-webgpu/src/backend_webgpu_test.ts#L243

How can the ready state be 1 (`HAVE_METADATA`) instead of 2 (`HAVE_CURRENT_DATA`) or higher here:

https://github.com/tensorflow/tfjs/blob/ca9799d16e23f7f31d4fdb15febb81d2eb4a2790/tfjs-core/src/ops/browser.ts#L112-L118

Am I misunderstanding anything? Thanks

cc @gyagp @qjia7 @shaoboyan

### PR DESCRIPTION
**Pull request by hujiajie**
`HTMLVideoElement.requestVideoFrameCallback()` provides a realiable way to guarantee the readiness of a video frame, whereas the browser may not update the ready state until a few frames later. In such cases, false alarms will be raised by the validation in `fromPixels()`.

Fixes #6577

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6626)

### CODE REVIEW

**Comment by hujiajie** [comment]
cc @nsthorat who initially authored this check, cc @gyagp who introduced rVFC in #6549

**Review by qjia7 — APPROVED** [review]
LGTM, thanks!

File: tfjs-backend-webgpu/src/from_pixels_webgpu_test.ts [hunk 1]
```diff
@@ -56,6 +56,11 @@ describeWebGPU('fromPixels', () => {
        if ('requestVideoFrameCallback' in video) {
          // tslint:disable-next-line:no-any
          await new Promise(go => (video as any).requestVideoFrameCallback(go));
+       } else {
+         // On mobile safari the ready state is ready immediately.
+         if (video.readyState < HTMLVideoElement.prototype.HAVE_CURRENT_DATA) {
```
Comment (gyagp) [inline tfjs-backend-webgpu/src/from_pixels_webgpu_test.ts:61]: WebGPU uses video.currentTime for check: https://github.com/gpuweb/cts/blob/0dd7405d1bfa950a71572aeb77aa38aacef6653d/src/webgpu/web_platform/util.ts#L48. 
WebGL uses a simpler solution as currently we have: https://github.com/KhronosGroup/WebGL/blob/main/sdk/tests/js/webgl-test-utils.js#L3182.

Comment (hujiajie) [reply tfjs-backend-webgpu/src/from_pixels_webgpu_test.ts]: Digging through the history of WebGL CTS, it seems awaiting `video.play()` + optional rVFC works fine across browsers and is preferred over `video.currentTime`, so I decided to drop the change here.

**Review by gyagp — APPROVED** [review]
LGTM

**Comment by gyagp** [comment]
@pyu10055 @lina128 @Linchenn PTAL!

**Review by pyu10055 — COMMENTED** [review]
thanks, does this change affects WebGL tests?

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6626#-:-N7b8P399OqMUuLMDT_V:bw3fegq)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @gyagp, @lina128, and @Linchenn)

**Review by lina128 — COMMENTED** [review]
Looks like this new API is not supported by Firefox and older Chrome/Safari yet, can we have a fallback? https://web.dev/requestvideoframecallback-rvfc/

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6626#-:-N7bLb9U4IfWFmo6hpNn:b-j045j9)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @gyagp and @Linchenn)

**Review by hujiajie — COMMENTED** [review]
Currently rVFC is only used in `from_pixels_webgpu_test.ts`, and the WebGPU backend is always incompatible with older Chrome/Safari. In this sense, rVFC is assumed to be supported by Chrome/Safari. It also sounds a fallback is redundant in Firefox because of its extra guarantee provided by `await video.play()` right above (https://github.com/KhronosGroup/WebGL/pull/3319).

WebGL tests seem unaffected by this. All are passing and I didn't grep any relevant case under the backend.

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6626#-:-N7iZ2HjBCVHZVKxylC6:b-hsy7ul)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @gyagp and @Linchenn)

**Review by pyu10055 — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6626)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @gyagp, @lina128, and @Linchenn)

**Review by gyagp — APPROVED** [review]
LGTM

**Review by Linchenn — APPROVED** [review]
LGTM!

**Pull request merged**